### DISCUSSION TITLE
Passing variables

### DISCUSSION THREAD

**Discussion by DenisHelloCharles** [anasoid/jmeter-as-code#170] [Q&A] [answered]
I'm wondering does it support next flow out of the box :
1. do login request, save token
2. do next request with token in headers, save several fields values from response
3. set saved values in 2nd request as parameters for 3rd request

Are there any examples on this?
Thanks!

**Comment by anasoid** [comment]
Hello,
There is no special feature on jmeter-as-code else writing jmeter (JMX) as code, as the scenario you ask  to do I think it's easy to do it in jmeter, yes you can write it in jmeter-as-code also.

**Reply by DenisHelloCharles** [reply]
Would you be so kind to share a simple java implementation example how this could be done please?

**Reply by anasoid** [reply]
Hello,

1- token is in header ?
2- where the value are in response ? and what is the format of returned data, json, HTML, ..

**Reply by DenisHelloCharles** [reply]
Hi,
The messages format is JSON.
1 - as an example could be taken Bearer token that comes in response body, than should be set in Authorization header
2 - take any one field value from message response body and set it as parameter in next request payload

**Comment by anasoid** [comment]
This example extract bearear token from json body and sent it in heaer and body

https://haquemousume.medium.com/jmeter-authorization-with-dynamic-access-token-bearer-and-generate-test-data-dynamically-83aa6df2ce56

**Reply by DenisHelloCharles** [reply]
I dont think its what I've been asking.
I'm not interested in Jmeter GUI mode. Thats easy to do, no issues. 

What I want to try is to use your java library to create java load script. If you could show how your framework might me used on Bearer token flow example it would be nice. 

If I undestand it correctly it might looks like:

```
TestPlanWrapper testPlan = TestPlanWrapper.builder()
        .addThread(ThreadGroupWrapper.builder()
            .addSampler(new LoginRequest())  // how to take "token" filed from response body here and put it as Authorization header for GetDashboardRequest? 

            .addSampler(new DashboardRequest())
            .build())
        .build();

    ApplicationTest applicationTest = new ApplicationTest(testPlanWrapper);

    applicationTest.run();
```

**Reply by DenisHelloCharles** [reply]
Its not clear how to use templates for LoginRequest and DashboardRequest
https://jmc.anasoid.org/docs/users/template

**Reply by anasoid** [reply]
if it's easy for you to do it in GUI mode, give me the jmx file and i will give you the java way

**Comment by DenisHelloCharles** [comment]
[Here it is](https://gist.github.com/DenisHelloCharles/6b5bf26d2792c40e6a07a958c516d643)

**Reply by anasoid** [reply]
are you familiar with maven or gradle ??

**Reply by DenisHelloCharles** [reply]
maven is preferable, thanks

**Comment by anasoid** [comment, marked as answer]
[denis-hello-charles.zip](https://github.com/anasoid/jmeter-as-code/files/7794101/denis-hello-charles.zip)
Execute 'mvn test' to generate jmx file.

I have do a small change for your test : 
1. Change test fragment by controller.
2. remove port and domain name from sampler as already present in default request. 

In attachment you will find a project with
**in source** : package one will contain all the test in one method (not preferable way for complex test).
                  package multi test is on many classes using also variable for token and template.
 **In test**  class GenerateJmxTest will generate in target folder :+1: 
1. **one.jmx** : test from one source class.
2. **multi.jmx** test from multi package
3. **multi-without-listener** variant of multi.jmx without listener

**Reply by DenisHelloCharles** [reply]
Hi, thats it! Many thanks!
